### ISSUE TITLE
Node process hangs when xray traced.

### ISSUE THREAD

**Issue by williamhoyle** [aws/aws-xray-sdk-node#241]
I have some jest tests that used to work under 2.3, but now hang. I've tracked the cause to a socket that is left open and referenced by the segment emitter.

It looks like this bug was introduced in [commit 7f9474034fabe508d23f2627dada42ae538712fd](https://github.com/aws/aws-xray-sdk-node/commit/7f9474034fabe508d23f2627dada42ae538712fd) when lazy socket initialisation was added. 

The code to `unref` the socket is run at the end of `segment_emitter.js`. After the change this no longer works because the socket does not exist until first send:

```javascript
if (SegmentEmitter.socket && (typeof SegmentEmitter.socket.unref === 'function'))
  SegmentEmitter.socket.unref();
```

**Comment by willarmiros** [comment]
Hi @williamhoyle,
Thank you for pointing this bug out. I'll be sure to include a fix in the next release since it's an easy change. I'd also be open to a PR if you have the time.

**Comment by willarmiros** [comment]
released in v3.0.0-alpha.2
